presto-main/src/test/java/io/prestosql/sql/planner/assertions/IdentityProjectionMatcher.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner.assertions;
+
+import io.prestosql.Session;
+import io.prestosql.cost.StatsProvider;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.sql.planner.plan.PlanNode;
+import io.prestosql.sql.planner.plan.ProjectNode;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.prestosql.sql.planner.assertions.MatchResult.NO_MATCH;
+import static io.prestosql.sql.planner.assertions.MatchResult.match;
+
+public class IdentityProjectionMatcher
```

Comment:
Can you help me see where this is used in the PR?

Reply:
It is used here https://github.com/prestosql/presto/blob/733de639a7639f3d8e26d5084a0ef5fa988e3d59/presto-main/src/test/java/io/prestosql/sql/planner/assertions/PlanMatchPattern.java#L342

it helps us ensure that above project node has only `IdentityProjections`